fakegato-schedule.js

```diff
@@ -0,0 +1,548 @@
+      ofs += 6;
+
+      // Temperature Levels
+      buf.writeUInt8(0xf4, ofs);
+      let tempLo = this.tempLo ? (this.tempLo * 2) : 0x80;
+      let tempHi = this.tempHi ? (this.tempHi * 2) : 0x80;
+      buf.writeUInt8(0x10, ofs+1); // unclear what this temperature indicates
+      buf.writeUInt8(0x10, ofs+2); // unclear what this temperature indicates
+      buf.writeUInt8(tempLo, ofs+3);
+      buf.writeUInt8(tempHi, ofs+4);
+      ofs += 5;
+
+      // Open window
+      buf.writeUInt8(0xf6, ofs);
+      if (this.openWindow) {
+        buf.writeUInt8(0x10, ofs+1); // this one bit is necessary
+      }
+      else {
+        buf.writeUInt8(0x00, ofs+1);
+      }
+      buf.writeUInt8(0x00, ofs+2); // ?
+      buf.writeUInt8(0x00, ofs+3); // ?
+      ofs += 4;
+
+      // Program (free day)
+      buf.writeUInt8(0x1a, ofs);
+      ofs += 1;
+      buf.write(this.programFree.hex, ofs, 'hex');
+      ofs += 8;
+
+      // Program (week)
+      buf.writeUInt8(0xfa, ofs);
+      ofs += 1;
+      buf.write(this.program1.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program2.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program3.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program4.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program5.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program6.hex, ofs, 'hex');
+      ofs += 8;
+      buf.write(this.program7.hex, ofs, 'hex');
+      ofs += 8;
+
+      //17 04 0a ????
+
+      //f3 38 19 00 00 ????
+
+      // trim buffer
+      let hexVal = buf.slice(0, ofs).toString('hex');
+      buf = Buffer.from(hexVal, 'hex');
+
+      this.log.debug(' - Data stream: %s (%d Byte)', hexVal, buf.length);
+      callback(null, buf.toString('base64'));
+    }
+
+    // callback function that is bound to SET ProgramCommand
+    cb_setProgramCommand(val, callback) {
+      this.log.debug('SET ProgramCommand:');
+      let buf = Buffer.from(val,'base64');
+      let length = buf.length;
+      let ofs = 0;
+
+      // flag variables for later commands
+      let setTemp = null;
+      let setEnableSchedule = null;
+      let setVacationMode = null;
+
+      let hexVal = buf.toString('hex');
+      this.log.debug(' - Data stream: %s', hexVal);
+
+      // parsing data stream
+      let opcode = 0;
+      while (ofs < length) {
+        opcode = buf.readUInt8(ofs);
+        ofs += 1;
+        switch(opcode) {
+          case 0x00: { // Start of Command
+            break;
+          }
+          case 0x06: { // End of Command
+            break;
+          }
+          case 0x10: { // Remove?
+            // do nothing
+            this.log.debug(' - Remove');
+            break;
+          }
+          case 0x11: { // Valve Protection
+            // do nothing
+            this.log.debug(' - valve protection');
+            break;
+          }
+          case 0x12: { // Temp Offset
+            this.tempOfs = buf.readInt8(ofs) / 10;
```

Comment:
Maybe an event could be emitted here, to make use of the temperature offset in external plugin code.
For now, it's just a fake setting.